cargo-spatial/Cargo.toml

```diff
@@ -0,0 +1,17 @@
+[package]
+name = "cargo-spatial"
+version = "0.1.0"
+authors = ["David LeGare <dlegare.1001@gmail.com>"]
+edition = "2018"
+
+[dependencies]
+glob = "0.2.11"
+log = "0.4.1"
+rand = "0.6.5"
+simplelog = "0.5.3"
+serde = { version = "1.0.38", features = ["derive"] }
+serde_json = "1.0.38"
+spatialos-sdk-code-generator = { path = "../spatialos-sdk-code-generator" }
```

Comment:
What if we wanted to allow users to plugin their own code generators here? 

I can imagine a use case where a user doesn't want our default implementation at all and wants to write their own layer. Could this be something we can specify in the `Spatial.toml` ?

Reply:
I'm pretty confident that's something we'll want to support. For example, for the Amethyst GDK we'll need to be able to generate the appropriate trait implementations for the relevant Amethyst traits, as well as any other glue code necessary for syncing SpatialOS data with the local world state, so custom code generation will be necessary. That said, I'd rather wait to add that functionality when I have a more concrete use case (i.e. when I'm starting to put together custom code generation for Amethyst), if you're cool with that.